atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventLog.java

```diff
@@ -143,12 +143,12 @@ private ClientLogEvents getEventsBetweenVersionsInternal(VersionBounds versionBo
         }
     }
 
-    @VisibleForTesting
-    LockWatchEventLogState getStateForTesting() {
+    @Unsafe
+    LockWatchEventLogState getStateForDiagnostics() {
         return ImmutableLockWatchEventLogState.builder()
                 .latestVersion(latestVersion)
-                .eventStoreState(eventStore.getStateForTesting())
-                .snapshotState(snapshot.getStateForTesting())
+                .eventStoreState(eventStore.getStateForDiagnostics())
+                .snapshotState(snapshot.getStateForDiagnostics())
                 .build();
     }
```

Comment:
Should I lock with the RW lock?

Reply:
yeah I'd recommend taking the read lock. Otherwise this may run at the same time as retention